src/Enum.php

```diff
@@ -52,64 +46,66 @@ public static function fromString(string $value): self
     {
         self::init();
 
-        if (array_key_exists($value, self::$values) === false) {
-            throw new InvalidValue($value, array_map(static fn (self $value) => $value->toString(), self::$values));
+        if (array_key_exists($value, self::$values[static::class]) === false) {
+            throw new InvalidValue(
+                $value,
+                array_map(
+                    static fn (self $value) => $value->toString(),
+                    self::$values[static::class]
+                )
+            );
         }
 
-        return self::$values[$value];
+        return self::$values[static::class][$value];
     }
 
-    /**
-     * @psalm-assert-if-true self::* $value
-     */
     public static function isValid(string $value): bool
     {
         self::init();
 
-        return array_key_exists($value, self::$values);
+        return array_key_exists($value, self::$values[static::class]);
     }
 
-    /**
-     * @psalm-return self::*
-     */
     public function toString(): string
     {
         return $this->value;
     }
 
     /**
-     * @psalm-param self::* $value
+     * @return static
      */
-    private static function get(string $value): self
+    protected static function get(string $value): self
     {
         self::init();
 
-        return self::$values[$value];
+        return self::$values[static::class][$value];
     }
 
     /**
      * @throws DuplicateValue
      */
-    private static function init(): void
+    protected static function init(): void
```

Comment:
same as above

Reply:
We need to access it in ExtendedEnum

Reply:
Hm, okay. This seems to be less problematic since `::init()` cannot break things i guess. Or do the maybe safe a state that the enum was already initialized and check  it here to be sure? Atm we only check it indirectly via the values property if there are already values registered. Should we harden this?

Reply:
i use I use the `fromString` method, so init can be private again.